### PR TITLE
feat(client): add multifile editor breadcrumbs

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Probably not exactly how we want the final style, but it bothered me a couple times.

### CODE REVIEW

File: client/src/templates/Challenges/components/challenge-title.css [hunk 1]
```diff
@@ -48,6 +47,24 @@
...
+  align-items: center;
+  justify-content: center;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  max-height: 25px;
+  flex-grow: 1;
+  flex-shrink: 2;
+  background-color: var(--quaternary-background);
+  overflow: hidden;
+  text-overflow: ellipsis;
```
Comment (Sembauke) [inline client/src/templates/Challenges/components/challenge-title.css:63]: `overflow: hidden` and `text-overflow ellipsis` is in the class twice.

Comment (ShaunSHamilton) [reply client/src/templates/Challenges/components/challenge-title.css:63]: You know, I think I have deleted that duplicate 3 times now...

Comment (Sembauke) [reply client/src/templates/Challenges/components/challenge-title.css]: I will keep an eye out for it :rofl:

File: client/src/templates/Challenges/components/bread-crumb.tsx [hunk 2]
```diff
@@ -22,20 +25,24 @@ function BreadCrumb({ block, superBlock }: BreadCrumbProps): JSX.Element {
         <span className='ellipsis'>
           {i18next.t(`intro:${superBlock}.title`)}
         </span>
-      </Link>
-      <div className='breadcrumb-center' />
-      {/* eslint-disable-next-line @typescript-eslint/ban-ts-comment */}
-      {/* @ts-ignore */}
+      </Link>{' '}
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/components/bread-crumb.tsx:28]:
```suggestion
      </Link>
```
Nitpick: this doesn't seem to change anything.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Thanks for adding these Shaun, they look good.
One thing I think could be improved is to have the colours alternate.  Instead of:

image

I think it would look better if went grey -> white -> grey, since the first arrow looks a bit odd. What do you reckon?

File: client/src/templates/Challenges/components/bread-crumb.tsx (hunk 2 above)
Comment (ShaunSHamilton) [reply client/src/templates/Challenges/components/bread-crumb.tsx:28]: I did not make that change. So, I think the Linter wanted it, but will do some testing later.

**Comment by ShaunSHamilton** [comment]
image

**Review by ojeytonwilliams — DISMISSED** [review]
LGTM :+1:

**Comment by nhcarrigan** [comment]
Might be a bit out of scope for the PR, but...

I'm not sure how I feel about having the challenge "title" in the breadcrumbs. What if we instead mirror our current UI and move the title into the description box?

image

**Comment by ShaunSHamilton** [comment]
> What if we instead mirror our current UI and move the title into the description box?

It would make things a lot simpler - BreadCrumbs component and css.

**Comment by nhcarrigan** [comment]
Assuming we make it look better than my two-second Dev Tools mockup there, I think it would be a more logical approach too.

**Comment by ShaunSHamilton** [comment]
@ojeytonwilliams If you are in agreement with Nich's suggestion, I will change this PR to match.

**Comment by ojeytonwilliams** [comment]
I'm happy with either approach, so please go ahead.

**Review by moT01 — COMMENTED** [review]
A couple thoughts just from looking at the images.

- Seems like that text in the breadcrumbs should be centered.
- The superBlock is bold and the block isn't - I feel like I would prefer no bold like the current ones.

I didn't actually run this to see it - or see what my thoughts implemented would look like - either of those may change my mind. It sounds like this is more of a initial implementation with the thought of maybe improving it later. So probably fine however you want to add it. Just some thoughts.

**Comment by nhcarrigan** [comment]
> Seems like that text in the breadcrumbs should be centered.

I agree - my mockup was more just to show "here's where I think the title should go" tbh

**Comment by ShaunSHamilton** [comment]
image

**Review by naomi-lgbt — APPROVED** [review]
image

Looks great. Nice work on this.

@ahmadabdolsaheb when you are back would you give this a look and merge if happy?

**Review by moT01 — APPROVED** [review]
I would be fine with this for now since it's a significant improvement. But some more thoughts:
Screen Shot 2021-07-21 at 5 51 24 PM

The arrow/box around the block title always seems to be quite a bit larger than the superblock box. I think equal sized would be best?

Since these are spanning the whole width of the window, they feel a little big. With the other layout, the breadcrumbs go in that small pane with the instructions - so they're a little more compact and maybe a little easier to understand what they are. Of course, those are partially cut off most of the time. I wouldn't worry about this for now, just a thought.

Like I said, I would be fine adding this. Looking good @ShaunSHamilton 👍 🎉

**Comment by ShaunSHamilton** [comment]
> The arrow/box around the block title always seems to be quite a bit larger than the superblock box. I think equal sized would be best?

Perhaps, but this was purposefully done with the current breadcrumbs.

> Since these are spanning the whole width of the window, they feel a little big.

Probably why the mock included the part in the breadcrumb.

---

As long as you think this is an overall improvement, we can discuss and refine later. 🙂

**Comment by ahmadabdolsaheb** [comment]
Sorry to be late at the party, we could merge this pr as is and iterate in the upcoming ones.
A simple breadcrumbs style with a way to show the progress at the end should do the trick, but I am open to suggestions.

Screen Shot 2021-07-23 at 1 04 00 PM

**Comment by nhcarrigan** [comment]
I'm okay with merging as-is, if you are. 🙂

**Comment by ahmadabdolsaheb** [comment]
Definitely, we can work on the UI if necessary down the line.

**Pull request merged**